blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/tasks/AlfrescoRepository.java

```diff
@@ -0,0 +1,170 @@
+import com.alfresco.apis.handler.Pair;
+import com.alfresco.apis.handler.auth.HttpBasicAuth;
+import com.alfresco.apis.model.NodeBodyUpdate;
+import com.alfresco.apis.model.NodeEntry;
+import com.alfresco.apis.model.RequestInclude;
+import com.alfresco.apis.model.RequestQuery;
+import com.alfresco.apis.model.ResultSetPaging;
+import com.alfresco.apis.model.ResultSetRowEntry;
+import com.alfresco.apis.model.SearchRequest;
+import com.google.common.collect.ImmutableMap;
+import com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState;
+import com.sphereon.libs.blockchain.commons.Digest;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class AlfrescoRepository {
+    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(AlfrescoRepository.class);
+
+    private static final String EXCEPTION_MESSAGE_FETCH_CONTENT = "An error occurred whilst fetching content: ";
+    private static final String EXCEPTION_MESSAGE_GET_NODE = "An error occurred whilst getting node %s: %s";
+
+    private final SearchApi alfrescoSearchApi;
+    private final ApiClient cmisApiClient;
+    private final NodesApi alfrescoNodesApi;
+
+    private String model;
+    private String registrationStateProperty;
+
+    public AlfrescoRepository(final SearchApi alfrescoSearchApi,
+                              final ApiClient cmisApiClient,
+                              final NodesApi alfrescoNodesApi,
+                              @Value("${sphereon.blockchain.agent.alfresco.query.model}") final String model,
+                              @Value("${sphereon.blockchain.agent.alfresco.query.registration-state:RegistrationState}") final String registrationStateProperty) {
+        this.alfrescoSearchApi = alfrescoSearchApi;
+        this.cmisApiClient = cmisApiClient;
+        this.alfrescoNodesApi = alfrescoNodesApi;
+        this.model = model;
+        this.registrationStateProperty = registrationStateProperty;
+    }
+
+    public List<ResultSetRowEntry> selectEntries(final AlfrescoBlockchainRegistrationState state) throws ApiException {
+        final SearchRequest request = new SearchRequest();
+        final RequestQuery query = new RequestQuery();
+        query.setLanguage(RequestQuery.LanguageEnum.AFTS);
+        query.setQuery('{' + model + '}' + registrationStateProperty + ':' + state.getKey());
+        request.setQuery(query);
```

Comment:
The order of request / query and then including it is probably more clear with one or two blank lines

Reply:
Wow, the whole ordering of this thing was wack. Changed it.